sdv/single_table/base.py

```diff
@@ -500,6 +506,75 @@ def get_info(self):
 
         return info
 
+    def _fit_constraint_column_formatters(self, data):
+        """Fit formatters for columns that are dropped by constraints before data processing."""
+        self._constraint_col_formatters = {}
+        self._constraint_col_dtypes = {}
+        primary_key = self.metadata.tables[self._table_name].primary_key
```

Comment:
Is it possible for the primary key to not be defined?

Reply:
The attribute is just `None` if the table doesn't have one set.